test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -1625,12 +1625,20 @@ INSTANTIATE_TEST_SUITE_P(IpVersionsClientTypeDeferredProcessing, ExtProcIntegrat
 TEST_P(ExtProcIntegrationTestUpstream, GetAndRespondImmediately_Upstream) {
   initializeConfig();
   HttpIntegrationTest::initialize();
-  auto response = sendDownstreamRequest(absl::nullopt);
+  auto response = sendDownstreamRequest([](Http::RequestHeaderMap& headers) {
+    // Setting this header to ensure that Envoy does not retry the request
```

Comment:
Is this comment accurate?

Setting x-envoy-retry-on header ----> ensure that Envoy *does* retry the request in case of a 5xx immediate response from the upstream

Reply:
ah i mean yes and no lol. "Setting this header (which causes Envoy to retry *usually* but this case is ext_proc is the one sending immediate response so) to ensure Envoy won't result in retry" is the accurate comment

Reply:
https://github.com/envoyproxy/envoy/pull/41231/commits/0805acdf22e70f0d5698e248940abc0d694ac055 here